sqlmesh/core/renderer.py

```diff
@@ -317,7 +317,9 @@ def _resolve_tables(
                 **table_mapping,
             }
             expand = set(expand) | {
-                name for name, snapshot in snapshots.items() if snapshot.is_embedded
+                name
+                for name, snapshot in snapshots.items()
+                if snapshot.is_embedded or not snapshot.categorized
             }
```

Comment:
Does this also render uncategorized grandparents? I suppose the `render_query` in 342 handles this recursively?

Reply:
Yes, because `expand` itself calls `render_query` which triggers the recursive sequence.